ocs/api_client.py

```diff
@@ -0,0 +1,255 @@
+    should implement all the methods for interacting with openshift cluster
+    """
+
+    @property
+    @abstractmethod
+    def name(self):
+        """Concrete class will have respective api-client name"""
+        pass
+
+    @abstractmethod
+    def api_get(self):
+        raise NotImplementedError("api_get method is not implemented")
+
+    @abstractmethod
+    def api_post(self):
+        raise NotImplementedError("api_post method is not implemented")
+
+    @abstractmethod
+    def api_delete(self):
+        raise NotImplementedError("api_delete method is not implemented")
+
+    @abstractmethod
+    def api_patch(self):
+        raise NotImplementedError("api_patch method is not implemented")
+
+    @abstractmethod
+    def api_create(self):
+        raise NotImplementedError("api_create method is not implemented")
+
+    @abstractmethod
+    def get_pods(self, **kwargs):
+        """
+        Because of discrepancies in IO format of each client api
+        leaving this to be implemented by specific client
+
+        Args:
+            **kwargs: ex: namespace='namespace',openshift namespace from
+                which we need pods
+
+        Raises:
+            NotImplementedError: if client has not implemented this function.
+
+        Returns:
+            pod_names (list): A list of pod names
+        """
+
+        raise NotImplementedError("get_pods method is not implemented")
+
+    @abstractmethod
+    def get_labels(self, pod_name, pod_namespace):
+        """
+        Get the openshift labels on a given pod
+
+        Args:
+            pod_name(str): Name of pod for which labels to be fetched
+            pod_namespace(str): namespace of pod where this pod lives
+
+        Raises:
+            NotImplementedError: if function not implemented by client
+
+        Returns:
+           dict: Labels associated with pod
+            """
+
+        raise NotImplementedError("get_labels method is not implemented")
+
+    @abstractmethod
+    def create_service(self, **kw):
+        """
+        Create an openshift service
+
+        Args:
+            **kw: ex body="content", namespace='ocnamespace'
+
+        Returns:
+            dict: Response from api server
+
+            Note: Returns could be tricky because it varies from client to
+            client. If its oc-cli then extra work needs to be done in the
+            specific implementation.
+        """
+
+        raise NotImplementedError("get_labels method is not implemented")
+
+
+# All openshift REST client specific imports here
+from oc import openshift_ops
+from openshift.dynamic import exceptions
+
+
+class OCRESTClient(APIClientBase):
+    """ All activities using openshift REST client"""
+
+    def __init__(self):
+        """TODO: get REST client instance from ctx which is shared globally"""
+        self.rest_client = openshift_ops.OCP()
+
+    @property
+    def name(self):
+        return "OCRESTClient"
```

Comment:
what about 
```python
return self.__class__.__name__
```

Reply:
Actually I was not sure whether config will have same parameter value as class name hence kept that string. However with this change user has to use same value as class name.

Done